### ISSUE TITLE
DisplayServer.tts_set_utterance_callback doesn't trigger when utterance finishes

### ISSUE THREAD

**Issue by nightblade9** [godotengine/godot#75192]
### Godot version

4.0.0 official binaries

### System information

Windows 10

### Issue description

Title says it all - adding a callback for TTS_UTTERANCE_ENDED or TTS_UTTERANCE_CANCELLED doesn't invoke the callable when speech finishes. This is on Windows 10.

### Steps to reproduce

Easy peasy, add this to any scene:

```gdscript
DisplayServer.tts_set_occurrence_callback(TTS_UTTARNENCE_STARTED, _done_speaking)
DisplayServer.tts_speak("hello, callback!")

func _done_speaking():
  print("DONE!!")
```

`Done!!` never prints.

### Minimal reproduction project

N/A

**Comment by nightblade9** [comment]
As-is, it appears to only be able to do this by writing some custom code that waits until `is_playing` is true and then again `false`. None of this is necessary on Linux, which makes the experience on Windows painful.

At present, it doesn't appear to be possible to play audio sequentially, without busy-waits that lock the UI. Is there a solution to this?

**Comment by nightblade9** [comment]
@Calinou this issue appears on Linux, as well.

**Comment by bruvzg** [comment]
`TTS_UTTERANCE_STARTED`, `TTS_UTTERANCE_ENDED`, and `TTS_UTTERANCE_CANCELED` callable's function should have one `int` argument, the utterance ID. Callable with the wrong argument count are never called.

**Comment by AThousandShips** [comment]
Confirming from the docs:

> Adds a callback, which is called when the utterance has started, finished, canceled or reached a text boundary.
>
> [TTS_UTTERANCE_STARTED](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-started), [TTS_UTTERANCE_ENDED](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-ended), and [TTS_UTTERANCE_CANCELED](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-canceled) callable's method should take one [int](https://docs.godotengine.org/en/stable/classes/class_int.html#class-int) parameter, the utterance id.
>
> [TTS_UTTERANCE_BOUNDARY](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-boundary) callable's method should take two [int](https://docs.godotengine.org/en/stable/classes/class_int.html#class-int) parameters, the index of the character and the utterance id.

**Comment by AThousandShips** [comment]
The reason this doesn't fire an error is that the callable are called deferred, and by default that doesn't show any errors in the message queue

**Comment by nightblade9** [comment]
> The reason this doesn't fire an error is that the callable are called deferred, and by default that doesn't show any errors in the message queue, but the documentation describes the callable

This seems like something that should be fixed. With other callbacks, you get an error if you specified the wrong number/types of arguments.

**Comment by AThousandShips** [comment]
Not all though, it depends

**Comment by nightblade9** [comment]
I will check if this works with a parameter in the callback.

> Not all though, it depends, but yes could be changed

If we leave this as-is, how can users who put the wrong signature in the TTS callback know why the callback isn't invoked?

**Comment by AThousandShips** [comment]
By reading the documentation, but it's not the universal, might be argued callbacks should throw errors when they are incorrect, but for a lot of cases they don't

**Comment by nightblade9** [comment]
> Details from the docs:
> 
> > Adds a callback, which is called when the utterance has started, finished, canceled or reached a text boundary.
> > [TTS_UTTERANCE_STARTED](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-started), [TTS_UTTERANCE_ENDED](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-ended), and [TTS_UTTERANCE_CANCELED](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-canceled) callable's method should take one [int](https://docs.godotengine.org/en/stable/classes/class_int.html#class-int) parameter, the utterance id.
> > [TTS_UTTERANCE_BOUNDARY](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#class-displayserver-constant-tts-utterance-boundary) callable's method should take two [int](https://docs.godotengine.org/en/stable/classes/class_int.html#class-int) parameters, the index of the character and the utterance id.

Yep, this works. Do you want to close this issue, or keep it open to address the lack of errors printed if the callback signature is incorrect?

**Comment by AThousandShips** [comment]
But if it's something that we want to add I can do it, investigated some places and adding errors and can open a PR

**Comment by MJacred** [comment]
@nightblade9: please rename this issue to `Not enough error handling for Callables` or sth. along the lines?
